src/composables/transactionList.ts

```diff
@@ -0,0 +1,324 @@
+  });
+
+  function setPendingTransactionSentByHash(address:Encoded.AccountAddress, hash: string) {
+    const index = transactions.value[address].pending[nodeNetworkId.value!]
+      ?.findIndex((transaction) => transaction.hash === hash);
+    if (index >= 0) {
+      transactions.value[address]
+        .pending[nodeNetworkId.value!][index][TX_DIRECTION.sent] = true;
+    }
+  }
+
+  async function waitTransactionMined(address: Encoded.AccountAddress, hash: string) {
+    try {
+      const aeSdk = await getAeSdk();
+      await aeSdk.poll(hash as Encoded.TxHash);
+      setPendingTransactionSentByHash(address, hash);
+    } catch (e) {
+      store.commit('removePendingTransactionByHash', { hash, network: nodeNetworkId.value });
+    }
+  }
+
+  function addCustomPendingTransactionForAccount(
+    address: Encoded.AccountAddress,
+    transaction: ITransaction,
+  ) {
+    if (!transactions.value[address]) {
+      transactions.value[address] = generateEmptyTransactionState();
+    }
+    transactions.value[address].pending[nodeNetworkId.value!] = [transaction];
+    waitTransactionMined(address, transaction.hash);
+  }
+
+  async function fetchTransactionsFromMiddleware(
+    transactionState: ITransactionsState,
+    address: Encoded.AccountAddress,
+    recent: boolean,
+    limit: number,
+  ) {
+    const { fetchFromMiddlewareCamelCased } = useMiddleware({ store });
+
+    const url = (recent || transactionState.nextPageUrl === '')
+      ? `/v2/accounts/${address}/activities?limit=${limit}`
+      : transactionState.nextPageUrl!;
+
+    try {
+      const { data, next } = await fetchFromMiddlewareCamelCased(url);
+      if (!recent) {
+        setTransactionsNextPage(address, next);
+      }
+      return data;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchPendingTransaction(
+    address: Encoded.AccountAddress,
+  ) {
+    const sdk = await getAeSdk();
+
+    try {
+      const fetchedPendingTransaction = (
+        await sdk.api.getPendingAccountTransactionsByPubkey(address)
+      );
+      const transactionState = getOrInitTransactionStateForAccount(address);
+
+      return JsonBig.parse(JsonBig.stringify(
+        fetchedPendingTransaction?.transactions || [],
+      ))
+        .filter((transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]
+          ?.find((tx) => tx?.hash === transaction?.hash))
+        .map((transaction: ITransaction) => ({ ...transaction, pending: true }));
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
+    try {
+      const response = await fetchJson(
+        `${nodeNetworkId.value}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
+      );
+      if (response.message) return [];
+      const tipWithdrawnTransactions = (response as any[]).map(({
```

Comment:
Instead of using `as any[]` here I suggest to type the source `response` constant.

Reply:
i added TODO for this type.